packages/utils/src/schema/toPathSchema.ts

```diff
@@ -73,16 +73,42 @@ function toPathSchemaInternal<T = any, S extends StrictRJSFSchema = RJSFSchema,
   }
 
   if (ITEMS_KEY in schema && Array.isArray(formData)) {
-    formData.forEach((element, i: number) => {
-      pathSchema[i] = toPathSchemaInternal<T, S, F>(
-        validator,
-        schema.items as S,
-        `${name}.${i}`,
-        rootSchema,
-        element,
-        _recurseList
-      );
-    });
+    const { items: schemaItems, additionalItems: schemaAdditionalItems } = schema;
+
+    if (Array.isArray(schemaItems)) {
+      formData.forEach((element, i: number) => {
+        if (schemaItems[i]) {
+          pathSchema[i] = toPathSchemaInternal<T, S, F>(
+            validator,
+            schemaItems[i] as S,
+            `${name}.${i}`,
+            rootSchema,
+            element,
+            _recurseList
+          );
+        } else if (schemaAdditionalItems) {
+          pathSchema[i] = toPathSchemaInternal<T, S, F>(
+            validator,
+            schemaAdditionalItems as S,
+            `${name}.${i}`,
+            rootSchema,
+            element,
+            _recurseList
+          );
+        }
```

Comment:
For the moment, let's add a `console.warn()` here to let users know that they have unhandled `formData`. Which means we want to update the test below with a check that `console.warn()` was called. This can be done with a `jest.spyOn()`

Reply:
Sure. Added a message. Let me know if there is a change required.

Reply:
@heath-freenome Should this log be moved up a level outside the for loop? We don't know how many items there will be in the array and if there are too many items, it might fill up the console.